setup.py

```diff
@@ -29,39 +28,14 @@ def read(*rnames):
         return f.read()
 
 
-def alltests():
-    import sys
-    import unittest
-    # use the zope.testrunner machinery to find all the
-    # test suites we've put under ourselves
-    import zope.testrunner.find
-    import zope.testrunner.options
-    here = os.path.abspath(os.path.join(os.path.dirname(__file__), 'src'))
-    args = sys.argv[:]
-    defaults = ["--test-path", here]
-    options = zope.testrunner.options.get_options(args, defaults)
-    suites = list(zope.testrunner.find.find_suites(options))
-    return unittest.TestSuite(suites)
-
-
-# PyPy cannot correctly build the C optimizations, and even if it
-# could they would be anti-optimizations (the C extension
-# compatibility layer is known-slow, and defeats JIT opportunities).
-py_impl = getattr(platform, 'python_implementation', lambda: None)
-pure_python = os.environ.get('PURE_PYTHON', False)
-is_pypy = py_impl() == 'PyPy'
-
-if pure_python or is_pypy:
-    ext_modules = []
-else:
-    ext_modules = [
-        Extension(
-            "zope.container._zope_container_contained",
-            [os.path.join("src", "zope", "container",
-                          "_zope_container_contained.c")],
-            include_dirs=['include'],
-        ),
-    ]
+ext_modules = [
+    Extension(
+        "zope.container._zope_container_contained",
+        [os.path.join("src", "zope", "container",
+                      "_zope_container_contained.c")],
+        include_dirs=['include'],
+    ),
+]
```

Comment:
Incidentally, does this mean you can't `pip install zope.container` from a source dist on a platform (*cough*Windows*cough*) with no C compiler?

Is that also the case for zope.interface?

I suppose we don't care, given that we're providing Windows wheels on PyPI.

Reply:
>  does this mean you can't pip install zope.container from a source dist on a platform (coughWindowscough) with no C compiler?

Yes, but:

1) Some of the zopefoundation repos have a custom `Extension` subclass that attempts to detect compiler errors and print a warning if compiling an extension fails (zope.interface does that). I'm not a big fan of that because (a) it too easily hides real issues; and (b) pip doesn't even show the message any more, so people don't know anything went wrong (pip only shows output if both building the wheel and installing directly from the sdist error out).
2) These days, getting a compiler for Windows is a simple download. It's no harder than it is to get a compiler for macOS. 
3) As mentioned, we do provide compiled binaries.